### ISSUE TITLE
`HttpClientFactory` passes wrong request options to `GuzzleHttp\Client`

### ISSUE THREAD

**Issue by schelmo** [rapidmail/rapidmail-apiv3-client-php#20]
the `version` field from the service is passed into the `GuzzleHttp\Client` Request options

so at the end we request http version `v1` which results in:
`HTTP/v1 is not supported by the cURL handler.`

https://docs.guzzlephp.org/en/stable/request-options.html#version

There are other fields which probably shouldnt go into guzzle request options:

> array:7 [▼ // vendor/rapidmail/rapidmail-apiv3-client-php/src/Http/HttpClientFactory.php:39
>   "base_uri" => "https://apiv3.emailsys.net/v1/"
>   **"throttle_interval" => 1
>   "throttle_requests_per_interval" => 10**
>   "version" => "v1"
>   **"auth" => array:2 [▶]**
>   "handler" => GuzzleHttp\HandlerStack {#1446 ▶}
>   "headers" => array:2 [▶]
> ]

setting `version` to null in
https://github.com/rapidmail/rapidmail-apiv3-client-php/blob/b354bf8973132271340098e885e4a1da8e86e8b9/src/Http/HttpClientFactory.php#L29-L39
is a workaround

**Comment by eth8505** [comment]
Thank you for reporting this bug. Will try to look into it next week.

**Comment by apaAB1D** [comment]
Another workaround is to downgrade guzzlehttp to 7.7.0.
Cheers

**Comment by eth8505** [comment]
@schelmo I have just released a new version 1.1.6 that unsets the version config before passing it to guzzle. Thank you for this.
I am aware this is a pretty dirty fix, but I don't see another way to maintain the current config structure and still allow for this.

My preferred long-term solution would be to move all guzzle-related config into a subkey of the config array, but that would break backwards compatibility. The workaround should do the job for now.

@apaAB1D Thank you for the hint about guzzle 7.7, I added this to the list of known issues in the readme, just in case.
